crosschain-resolver/test/testResolver.ts

```diff
@@ -0,0 +1,135 @@
+      return {
+        statusCode: response.statusCode,
+        statusMessage: response.ok ? 'OK' : response.statusCode.toString(),
+        body: ethers.toUtf8Bytes(JSON.stringify(response.body)),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      };
+    });
+    const l1VerifierFactory = await ethers.getContractFactory(
+      'L1Verifier',
+      signer
+    );
+    verifier = await l1VerifierFactory.deploy(['test:']);
+
+    const testL2Factory = await ethers.getContractFactory(
+      'OwnedResolver',
+      signer
+    );
+    l2contract = await testL2Factory.deploy();
+
+    const testL1Factory = await ethers.getContractFactory(
+      'L1Resolver',
+      signer
+    );
+    target = await testL1Factory.deploy(await verifier.getAddress(), await l2contract.getAddress());
+
+    // Mine an empty block so we have something to prove against
+    await provider.send('evm_mine', []);
+  });
+
+  it("should test ETH Address", async() => {
+    const addr = '0x5A384227B65FA093DEC03Ec34e111Db80A040615'
+    await l2contract.clearRecords(node)
+    await l2contract['setAddr(bytes32,address)'](node, addr)
+    const result = await l2contract['addr(bytes32)'](node)
+    expect(ethers.getAddress(result)).to.equal(addr);
+    await provider.send("evm_mine", []);
+    const result2 = await target['addr(bytes32)'](node, { enableCcipRead: true })
+    expect(result2).to.equal(addr);
+  })
+  it("should test non ETH Address", async() => {
+    const addr = '0x76a91462e907b15cbf27d5425399ebf6f0fb50ebb88f1888ac'
+    const coinType = 0 // BTC
+    await l2contract.clearRecords(node)
+    await l2contract['setAddr(bytes32,uint256,bytes)'](node, coinType, addr)
+    const result = await l2contract['addr(bytes32,uint256)'](node, 0)
+    expect(result).to.equal(addr);
+    await provider.send("evm_mine", []);
+    const result2 = await target['addr(bytes32,uint256)'](node, coinType, { enableCcipRead: true })
+    expect(result2).to.equal(addr);
+  })
+
+  it("should test text record", async() => {
+    const key = 'name'
+    const value = 'nick.eth'
+    await l2contract.clearRecords(node)
+    await l2contract.setText(node, key, value)
+    await provider.send("evm_mine", []);
+    const result = await l2contract.text(node, key)
+    expect(result).to.equal(value);
+    const result2 = await target.text(node, key, { enableCcipRead: true })
+    expect(result2).to.equal(value);
+  })
+
+  it("should test contenthash", async() => {
+    const contenthash = '0xe3010170122029f2d17be6139079dc48696d1f582a8530eb9805b561eda517e22a892c7e3f1f'
+    await l2contract.clearRecords(node)
+    await l2contract.setContenthash(node, contenthash)
+    await provider.send("evm_mine", []);
+    const result = await l2contract.contenthash(node)
+    expect(result).to.equal(contenthash);
+    const result2 = await target.contenthash(node, { enableCcipRead: true })
+    expect(result2).to.equal(contenthash);
+  })
+});
```

Comment:
Some tests that demonstrate the behaviour when records don't exist would be valuable.